chat.go

```diff
@@ -61,40 +62,8 @@ type ChatCompletionRequest struct {
 type FunctionDefine struct {
 	Name        string `json:"name"`
 	Description string `json:"description,omitempty"`
-	// it's required in function call
-	Parameters *FunctionParams `json:"parameters"`
```

Comment:
go-openai keeps compatible. This is breaking changes. So why not add a field like `ParametersRaw`?
Note: You need to implement `MarshalJSON` func in `FunctionDefine` since the "parameters" tag is duplicated.

Reply:
heh, yeah, was a bit sad to see the release go out before I had a chance to upstream this. I agree it's a bit undesirable to have a breaking change like this, even if the release was only out for a few hours. 

It should be relatively straightforward to implement a custom marshaler here, but I think it's worth having the larger conversation about how this library intends to handle the Parameters field moving forward - will this library implement (and maintain) its own JSONSchemaDefine (and related structs) or not?

If so - then ParametersRaw is the only path forward, and I will write the custom functions. GPT-4 gets us pretty close I think: https://chat.openai.com/share/bec0bb09-41f5-4746-8b21-8e40087509b1

If not - then the cleanest path is to deprecate Parameters as an argument and ask people to use ParametersRaw. That's not the end of the world, but it would be much more intuitive to match the API schema. 

My view is that the JSON Schema definition is too large to reasonably support in this library and is better suited to be handled in a specialized library. The problem with supporting a subset is that once somebody hits an edge case, then they have to completely rewrite their code here to use either the raw bytes or another library's struct that can serialize properly.

Reply:
I have been testing out this PR, and I like it a lot! I think it's a lot smoother and easier to work with, and probably easier for new devs, too (for quick testing you can just paste in a schema as a text string, instead of building one from nested Go structs!)

I think this is the way we should do it.

Reply:
I think it should also be marked as deprecated by prepending `// Deprecated:` at the very last line of comment and stated out that users should go and use `ParametersRaw` instead.